src/libstore/misc.cc

```diff
@@ -200,6 +200,40 @@ void Store::queryMissing(const std::vector<DerivedPath> & targets,
             auto drv = make_ref<Derivation>(derivationFromPath(bfd.drvPath));
             ParsedDerivation parsedDrv(StorePath(bfd.drvPath), *drv);
 
+            if (!knownOutputPaths && settings.useSubstitutes && parsedDrv.substitutesAllowed()) {
+                // Try to realise the static output hashes.
+
+                // TODO: I'm not sure this logic is quite right, how do we
+                // ensure all the needed paths are present?
+
+                // TODO: Should we also do this->queryRealisation?
```

Comment:
Not sure whether this code path would be hit if this->queryRealisation could succeed. I figure probably not.

Reply:
It won't indeed. If we know the realisation, then `queryPartialDerivationOutputMap` will honor that and give us a path for the output